contractcourt/commit_sweep_resolver_test.go

```
@@ -0,0 +1,186 @@
+func newMockSweeper() *mockSweeper {
+	return &mockSweeper{
+		sweptInputs: make(chan input.Input),
+	}
+}
+
+func (s *mockSweeper) SweepInput(input input.Input,
+	feePreference sweep.FeePreference) (chan sweep.Result, error) {
+
+	s.sweptInputs <- input
+
+	result := make(chan sweep.Result, 1)
+	result <- sweep.Result{
+		Tx: &wire.MsgTx{},
+	}
+	return result, nil
+}
+
+func (s *mockSweeper) CreateSweepTx(inputs []input.Input, feePref sweep.FeePreference,
+	currentBlockHeight uint32) (*wire.MsgTx, error) {
+
+	return nil, nil
+}
+
+var _ UtxoSweeper = &mockSweeper{}
+
+// TestCommitSweepResolverNoDelay tests resolution of a direct commitment output
+// unencumbered by a time lock.
+func TestCommitSweepResolverNoDelay(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	res := lnwallet.CommitOutputResolution{
+		SelfOutputSignDesc: input.SignDescriptor{
+			Output: &wire.TxOut{
+				Value: 100,
+			},
+		},
+	}
+
+	ctx := newCommitSweepResolverTestContext(t, &res)
+	ctx.resolve()
+
+	ctx.notifier.confChan <- &chainntnfs.TxConfirmation{}
+
+	// No csv delay, so the input should be swept immediately.
+	<-ctx.sweeper.sweptInputs
+
+	ctx.waitForResult()
+}
+
+// TestCommitSweepResolverDelay tests resolution of a direct commitment output
+// that is encumbered by a time lock.
+func TestCommitSweepResolverDelay(t *testing.T) {
+	t.Parallel()
+	defer timeout(t)()
+
+	res := lnwallet.CommitOutputResolution{
+		SelfOutputSignDesc: input.SignDescriptor{
+			Output: &wire.TxOut{
+				Value: 100,
+			},
+		},
+		MaturityDelay: 3,
+	}
+
+	ctx := newCommitSweepResolverTestContext(t, &res)
+	ctx.resolve()
+
+	ctx.notifier.confChan <- &chainntnfs.TxConfirmation{
+		BlockHeight: testInitialBlockHeight - 1,
+	}
+
+	// Notify initial block height. The csv lock is still in effect, so we
+	// don't expect any sweep to happen yet.
+	ctx.notifyEpoch(testInitialBlockHeight)
+
+	select {
+	case <-ctx.sweeper.sweptInputs:
+		t.Fatal("no sweep expected")
+	case <-time.After(time.Second):
```

can probably lower this to the order of milliseconds

---

isn't that possibly causing travis flakes? I thought that because everything is executed in parallel, it is ok to take a more generous timeout

---

But this is the happy flow, so if the timeout is "too low" it will just silently succeed. In the event a bug is introduced such that inputs get swept, I think it will be caught in most cases also with a lower timeot.

---

Sounds that lowering it increases the risk of something slipping through. With parallel execution, I think there isn't much to gain? Lowered to 100 ms nevertheless.